spec.bs

```diff
@@ -747,19 +789,42 @@ Issue(43): This should be moved to the Shared Storage spec.
 partial interface SharedStorageWorkletGlobalScope {
   readonly attribute PrivateAggregation privateAggregation;
 };
+
+dictionary SharedStoragePrivateAggregationConfig {
+  USVString contextId;
+};
+
+partial dictionary SharedStorageRunOperationMethodOptions {
+  SharedStoragePrivateAggregationConfig privateAggregationConfig;
+};
 </xmp>
 
 The {{WindowSharedStorage}}'s {{WindowSharedStorage/run()}} method steps are
-modified in two ways. First, add the following step in the nested scope just
-after "Let |operation| be |operationMap|[|name|]." (renumbering later steps as
-appropriate):
+modified in three ways. First, add the following steps just after step 2 ("If
+{{Worklet/addModule()}} has not yet been called, ..."), renumbering later steps
+as appropriate:
 <div algorithm="shared-storage-run-monkey-patch-1">
-2. Let <var ignore>batchingScope</var> be a new [=batching scope=].
-1. Let <var ignore>debugScope</var> be a new [=debug scope=].
+3. Let |contextId| be null.
+1. If |options|["`privateAggregationConfig`"] [=map/exists=]:
+    1. If |options|["`privateAggregationConfig`"]["`contextId`"] [=map/exists=],
+        1. Set |contextId| to
+            |options|["`privateAggregationConfig`"]["`contextId`"].
+        1. If |contextId|'s [=string/length=] is greater than 64, return [=a
```

Comment:
Shall we move this step above before setting the contextId?

Reply:
I've kept it (now in the separate algorithm) just to avoid saying the long `options["privateAggregationConfig"]["contextId"]` again, but I can switch it if you'd still prefer

Reply:
LGTM!